File: typescript/src/examples/solana/createAccount.ts
// Usage: pnpm tsx src/examples/solana/createAccount.ts

import { config } from "dotenv";

import { CdpClient } from "../../client/cdp";

/**
 * This example shows how to create a new Solana account
 */
async function main() {
  config();

  const cdp = new CdpClient();

  try {
    const account = await cdp.solana.createAccount();
    console.log("Successfully created Solana account:", account);
  } catch (error) {
    console.error("Error creating Solana account:", error);
  }
}

main().catch(console.error);

File: python/cdp/examples/solana/create_account.py
# Usage: uv run python cdp/examples/solana/create_account.py

import asyncio

from dotenv import load_dotenv

from cdp import CdpClient


async def main():
    """Contains main function for the Solana account script."""
    load_dotenv()

    async with CdpClient() as cdp:
        solana_account = await cdp.solana.create_account()
        print(f"Successfully created SOL account: {solana_account.address}")


# Run the async function
asyncio.run(main())
